src/harness/reference_models/inter_sas_duplicate_grant/inter_sas_duplicate_grant.py

```diff
@@ -0,0 +1,95 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""
+==================================================================================
+  This is a subset of the pre-IAP reference model which implements inter-SAS
+  duplicate grant removal. If a CBSD has registered with multiple SASs then has
+  overlapping grants then the overlapping grants will be removed from the FAD
+  objects. The main function is interSasDuplicateGrantReferenceModel().
+==================================================================================
+"""
+
+
+def interSasDuplicateGrantReferenceModel(sas_uut_fad, sas_test_harness_fads):
+  """ Removes duplicate grants from FAD objects of SAS UUT and SAS test harnesses.
+
+  Checks if a CBSD is registered with more than one SAS and removes the grants of the
+  CBSD if the grant frequencies overlap either fully or partially. If all grants are
+  removed from a CBSD then the CBSD is removed from the FAD Object.
+
+  Args:
+    sas_uut_fad: A FullActivityDump object containing the FAD records of SAS UUT.
+    sas_test_harness_fads: A list of FullActivityDump objects containing the FAD records
+      from SAS test harnesses.
+  """
+
+  # Creating list of cbsds from SAS UUT FAD object and SAS Test Harness FAD object
+  cbsd_list = []
+  for cbsds in sas_uut_fad.getCbsdRecords():
```

Comment:
Since all that matters is CBSD reference ID (see Kate's comment below) you can just keep track of the number of occurrences of a given ID (in a dictionary) and only include CBSDs with a single occurrence in the output.

Reply:
Thanks.   The logic has been updated. The input parameters are modified and no output   is provided.